### PR TITLE
Feature/email service

### PR DESCRIPTION
**Pull request by jeremNoel**
new PR, follow to https://github.com/Mak-API/website/pull/20

### CODE REVIEW

**Review by mehdou92 — APPROVED** [review]

**Review by benKazmierczak — APPROVED** [review]

**Review by Backins — APPROVED** [review]
I approve, but check Codacy : 
* don't use "else" with just a return. Use this : 
```
if(condition){
return true;
}
return false;
```

* getenv is discouraged. Please check if it's the best solution

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -17,6 +18,14 @@
  */
 class UserController extends AbstractController
 {
+
+    private $emailService;
```
Comment (ivanPrunier) [inline src/Controller/UserController.php:25]: On top of class variables like this, a phpdoc comment `@var EmailService` doesn't eat bread (lol)

File: src/Controller/UserController.php [hunk 2]
```diff
@@ -29,33 +38,54 @@ public function index(UserRepository $userRepository): Response
...
 
+
+
             return $this->redirectToRoute('app_user_index');
         }
 
         return $this->render('user/new.html.twig', [
             'user' => $user,
-            'form' => $form->createView(),
+            'form' => $form->createView()
```
Comment (ivanPrunier) [inline src/Controller/UserController.php]: In general it's better to leave the last comma (usually called trailing comma):
right now if I add a new key to the array, I have to modify two lines: the currently last one to add the comma, and the next one to add my key/value. If you leave the trailing comma, I'll only have to add a line ! It makes PRs a bit easier to read !

File: src/Entity/User.php [hunk 3]
```diff
@@ -27,6 +27,7 @@ class User implements UserInterface
      */
     const ROLE_USER = 'ROLE_USER';
     const ROLE_ADMIN = 'ROLE_ADMIN';
+    const ROLE_UNVERIFIED = 'ROLE_UNVERIFIED';
```
Comment (ivanPrunier) [inline src/Entity/User.php:30]: For now we can keep the ROLE_UNVERIFIED but in the long run I think we will need to separate the notion of role and the notion of status: when you create an account you have a ROLE_USER and a STATUS_UNVERIFIED, then your status evolves but you're still a ROLE_USER. 
As I said not important right now but let's keep that in mind !

File: src/Security/LoginFormAuthenticator.php [hunk 4]
```diff
@@ -89,6 +101,17 @@ public function onAuthenticationSuccess(Request $request, TokenInterface $token,
         return new RedirectResponse($this->router->generate('app_default_index'));
     }
 
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception)
+    {
+            //Catch code, if it's the custom 410 error, call the function for rendering the template
+          if($exception->getCode() == 410){
```
Comment (ivanPrunier) [inline src/Security/LoginFormAuthenticator.php]: ===

File: src/Service/EmailService.php [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+use phpDocumentor\Reflection\Types\Boolean;
+
+class EmailService {
+
+    /**
+     * @var \Swift_Mailer
+     *
+     */
+    private $mailer;
+    private $templating;
```
Comment (ivanPrunier) [inline src/Service/EmailService.php:20]: phpdoc blocs on the other vars as well ;)

File: src/Service/EmailService.php [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+use Aws\Exception\AwsException;
+use Doctrine\Common\Persistence\ObjectManager;
+use mysql_xdevapi\Exception;
+use phpDocumentor\Reflection\Types\Boolean;
+
+class EmailService {
+
+    /**
+     * @var \Swift_Mailer
+     *
```
Comment (ivanPrunier) [inline src/Service/EmailService.php:17]: useless line :D

File: src/Service/UserService.php [hunk 7]
```diff
@@ -8,10 +8,77 @@
...
+            $user->setVerified(true);
+            $this->manager->flush();
+            return ["isVerified" => true,
+                "login" => $user->getEmail()];
+        } else if($user && $user->getVerified()) {
+            throw new \Exception('Something went wrong!');
+        } else if(!$user) {
+            throw new \Exception('Something went wrong!');
+        } else {
+            throw new \Exception('Something went wrong!');
```
Comment (ivanPrunier) [inline src/Service/UserService.php]: Not important right now but we should have separate exceptions to be able to know exactly what happened !

**Review by ivanPrunier — DISMISSED** [review]
Small changes to do so if you can treat them before merging it'd be perfect!

File: src/Service/EmailService.php (hunk 6 above)
Comment (jeremNoel) [reply src/Service/EmailService.php:17]: Done

File: src/Service/EmailService.php (hunk 5 above)
Comment (jeremNoel) [reply src/Service/EmailService.php:20]: I let you check the last commit that i've done.

File: src/Security/LoginFormAuthenticator.php (hunk 4 above)
Comment (jeremNoel) [reply src/Security/LoginFormAuthenticator.php]: Done

File: src/Entity/User.php (hunk 3 above)
Comment (jeremNoel) [reply src/Entity/User.php:30]: Note

File: src/Controller/UserController.php (hunk 2 above)
Comment (jeremNoel) [reply src/Controller/UserController.php]: Done, thanks for the tips :)

File: src/Controller/UserController.php (hunk 1 above)
Comment (jeremNoel) [reply src/Controller/UserController.php:25]: Done

File: src/Service/EmailService.php (hunk 5 above)
Comment (jeremNoel) [reply src/Service/EmailService.php:20]: k, talk about this later

**Comment by jeremNoel** [comment]
as @ivanPrunier said on slack, I can merge this PR.

**Pull request merged**